src/access_list.rs

```diff
@@ -59,12 +54,63 @@ impl AccessListInspector {
         });
         AccessList(items.collect())
     }
+
+    /// Collects addresses which should be excluded from the access list.
+    ///
+    /// Those include caller, callee and precompiles.
+    fn collect_excluded_addresses<DB: Database>(&mut self, context: &EvmContext<DB>) {
+        let from = context.env.tx.caller;
+        let to = if let TxKind::Call(to) = context.env.tx.transact_to {
+            to
+        } else {
+            let nonce = context.journaled_state.account(from).info.nonce;
+            from.create(nonce)
```

Comment:
let's add sanity note here that this will be a create call and we should exclude the created address